### PR TITLE
Minor changes to testing setup

### PR DESCRIPTION
**Pull request by dghubble**
Builds atop great testing work in #11 by @mcuadros and makes a few minor changes.

* Use tlsutil.TLSInfo and ServerConfig from matchbox package
* Callers goroutine the server rather than doing it in the Start function itself
* Allow server and client TLS credentials to be passed in to FixtureServer
* Stop is a NOP if Start has not yet been called
* ~Don't hold a listener in the struct, just use a constant address (possibly temporary, try to find races and report upstream)~

### CODE REVIEW

**Comment by dghubble** [comment]
There do seem to be races to do with the gRPC package's server.Stop and address:port release, I see the address already bound error in 1 of every ~200 test runs.

Review of Linux Programming Interface shows this is fairly foundational to socket close (search WAIT_TIME and SO_REUSEADDR) and several grpc upstream discussions just mention randomizing ports when setting up and tearing down tests. In that light, let's stick with Maximo's random ports and not worry about this.

**Comment by mcuadros** [comment]
Awesome, but if you get back to the random ports... you will need to keep the listener or not?

**Comment by dghubble** [comment]
Yes, keeping the listener in the struct. I think I was just pushing as you commented.

File: matchbox/fixture_test.go [hunk 1]
```diff
@@ -63,183 +93,20 @@ CA
...
+		s.ClientTLS.Cert,
+		s.ClientTLS.Key,
+		s.ClientTLS.CA,
+		hcl)
 }
 
-func testServerTLSConfig() (*tls.Config, error) {
-	cert, err := tls.X509KeyPair(fixtureServerCertPEMBlock, fixtureServeKeyPEMBlock)
+// mustFile wraps a call to ioutil.ReadFile and panics if the error is non-nil.
+func mustFile(filename string) []byte {
```
Comment (mcuadros) [inline matchbox/fixture_test.go]: a better name will be MustReadFile?

Comment (dghubble) [inline matchbox/fixture_test.go]: :+1:

**Pull request merged**